pkg/operator/resource/resourceapply/core.go

```diff
@@ -150,6 +154,7 @@ func ApplyServiceAccount(client coreclientv1.ServiceAccountsGetter, recorder eve
 func ApplyConfigMap(client coreclientv1.ConfigMapsGetter, recorder events.Recorder, required *corev1.ConfigMap) (*corev1.ConfigMap, bool, error) {
 	existing, err := client.ConfigMaps(required.Namespace).Get(required.Name, metav1.GetOptions{})
 	if apierrors.IsNotFound(err) {
+		resourcemerge.CleanMetaLabelsAnnotations(&required.ObjectMeta)
```

Comment:
mutating the input is bad. We need a deep copy (or shallow copy of at least the root and ObjectMeta).

Reply:
I went with the DeepCopy() should we ever want to mutate any other fields of the actual object as well

Reply:
performance here does not matter anyway